js/apps/admin-ui/src/user/form-state.ts

```diff
@@ -17,7 +17,9 @@ export function toUserFormFields(data: UserRepresentation): UserFormFields {
 
 export function toUserRepresentation(data: UserFormFields): UserRepresentation {
   const username = data.username?.trim();
-  const attributes = keyValueToArray(data.attributes);
+  const attributes = Array.isArray(data.attributes)
```

Comment:
Under what circumstances is `data.attributes` not an array?

Reply:
@edewit would you know this? Thanks!

Reply:
it's not an array when you are using user profile

Reply:
Ah I see, this is because we're calling the t[his piece of code](https://github.com/keycloak/keycloak/blob/c15753266ff7e86af82ff2a4a82fc26f0cbcc16f/js/apps/admin-ui/src/user/EditUser.tsx#L115) conditionally?

Reply:
no, it's because the components that render the user profile already put the values in the right form e.g.

```json
{
   test: "value"
}
```

Reply:
I don't mind having a different value in the forms depending on whether user profile is enabled, but we should also ensure that the type is actually correct. In this case the type of `data.attributes` is either `{ key: string; value: string }[]` or `{  [key: string]: string | string[] }`, but the type does not reflect that.

Personally, I think we're making a mistake not fixing the biggest problem here, which is that the returned value from the REST API is very cumbersome. Attributes should always be `{ key: string, value: string }` or even have a `type` so we don't [have to guess (wrongly)](https://github.com/keycloak/keycloak/blob/c15753266ff7e86af82ff2a4a82fc26f0cbcc16f/js/apps/admin-ui/src/user/UserProfileFields.tsx#L119-L120) what the original intention was.

But this is not something we can or should fix right now, we just need to get this bug resolved. So the correct type for [`UserFormFields`](https://github.com/keycloak/keycloak/blob/a9fc29cac524fef8eab86f7b54854a20e3275a0f/js/apps/admin-ui/src/user/form-state.ts#L8-L10) should be:

```ts
export type UserFormFields = Omit<
  UserRepresentation,
  "attributes" | "userProfileMetadata"
> & {
  attributes?: KeyValueType[] | Record<string, string | string[]>;
};
```